<a href="https://www.geneweaver.org">
    <h1 align="center">
        <img src="https://geneweaver.org/static/images/GW2-logo.png" alt="GeneWeaver Logo" style="vertical-align: top">
        GeneWeaver
    </h1>
</a>
<p align="center">
    <em>A system for the integration and analysis of heterogeneous functional genomics data.</em>
</p>

---
**Web Application**: 
[https://www.geneweaver.org](https://www.geneweaver.org)

**API Documentation**:
[https://geneweaver.jax.org/api/docs](https://geneweaver.jax.org/api/docs)
---

<div class="grid cards" markdown>

-   :material-clock-fast:{ .lg .middle } __Get Going Quickly__

    ---

    Architected to make it easy to get started, whichever way makes sense for you.

    [:octicons-arrow-right-24: Jump Into Using The Website](/getting-started/web-application/)

    [:octicons-arrow-right-24: Download the Geneweaver CLI](/reference/command-line/)

    [:octicons-arrow-right-24: Build With Our Python Packages](/reference/available-packages/)

-   :simple-jupyter:{ .lg .middle } __Example Workflows__

    ---

    Follow real examples of how GeneWeaver's tools will be used to do analysis.

    [:octicons-arrow-right-24: NCI-60 Workflow](/tutorial/nci_60_example_01/)

    [:octicons-arrow-right-24: Authenticating With The Client Library](/tutorial/geneweaver_client_login/)

-   :material-api:{ .lg .middle } __API Standards__

    ---

    Developed to be consistent with our open API Standards.

    [:octicons-arrow-right-24: Standards Documentation](/reference/api-standards/)

-   :material-scale-balance:{ .lg .middle } __Open Source, Apache License 2.0__

    ---

    GeneWeaver is open source and available on [GitHub](https://github.com/orgs/TheJacksonLaboratory/repositories?q=geneweaver).

    [:octicons-arrow-right-24: License](https://github.com/TheJacksonLaboratory/geneweaver-docs/blob/main/LICENSE)

</div>


## What Is GeneWeaver?
**[GeneWeaver](http://www.geneweaver.org)** is a web application for the integrated
cross-species analysis of functional genomics data from heterogeneous sources. The
application consists of a large database of gene sets curated from multiple public data
resources and curated submissions, along with a suite of analysis tools designed to
allow flexible, customized workflows through web-based interactive analysis or scripted
API driven analysis. Gene sets come from multiple widely studied species and include
ontology annotations, brain gene expression atlases, systems genetic study results, gene
regulatory information, pathway databases, drug interaction databases and many other
sources. Users can retrieve, store, analyze and share gene sets through a graded access
system. Gene sets and analysis results can be stored, shared and compared privately,
among user defined groups of investigators, and across all users. Analysis tools are
based on combinatorics and statistical methods for comparing, contrasting and
classifying gene sets based on their members.

## What Is a Gene Set?
Each **"gene set"** contains a list of genomic features, free text descriptive content,
ontology annotations and gene association scores. Genomic features are mapped within and
across multiple species. Currently, 10 species are supported, *Mus musculus, Homo
sapiens, Rattus norvegicus, Danio rerio, Drosophilia melanogaster, Macaca mulatta,
Caenorhabditis elegans, Saccharomyces cervisiae, Gallus gallus, Canis familiaris.*
Additional species are added in response to community request.

## What Can You Do With GeneWeaver?
**[GeneWeaver](http://www.geneweaver.org)** allows users to integrate these diverse
functional genomics data across species, tissue and experimental platform to address
questions about the relations among genes and biological functions. Applications include
the prioritization of gene-disease associations from multiple evidence sources, the
classification and comparison of biological functions based on biological substrates,
and the identification of similar genes based on function. Cross species analysis
enables the discovery of conserved mechanisms of biological functions, and the discovery
of divergent functions served by conserved biological mechanisms. 

## What is this Documentation?
This documentation provides an introduction to the system and ways of
working with it, explanations of core concepts, tutorials of general utility,
as well as detailed reference documentation for the various components and processes
that make up the system.
